controllers/metal3machine_controller.go

```diff
@@ -244,6 +249,7 @@ func (r *Metal3MachineReconciler) reconcileNormal(ctx context.Context,
 	// Make sure that the metadata is ready if any
 	err := machineMgr.AssociateM3Metadata(ctx)
 	if err != nil {
+		machineMgr.SetConditionMetal3MachineToFalse(infrav1.KubernetesNodeReadyCondition, infrav1.AssociateM3MetaDataFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
```

Comment:
same here as above. I always thought for some reason the conditions has to be marked as true

Reply:
KubernetesNodeReadyCondition as name says it should be ready, when it is false it will show a reason why it is false.